### PR TITLE
Release 4.2.0

### PR DESCRIPTION
**Pull request by ggabernet**
- Adds TRUST4 support
- Adds options to remove V and C primer region by length when primer sequences are unknown.
- Adds option to save non-productive sequences after IgBlast alignment.
- Updated nf-core template to v3.0.2
- Deprecated isotype_column param in favor of annotating C region or C primer information in the c_call column when c_call is NA.
- Avoids saving pRESTO intermediate files in results directory to save space.
- Simplified pRESTO Maskprimers score and Maskprimers extract processes.
- Fix clustersets approach, consider both reads clusters annotation.
- Allow for partial alignments in MakeDB for mouse TCR data, to bypass junction check for TRAV germlines with additional positions (e.g. 84A).
- Fix parse logs merging dataframes using pandas join to produce logs even when there are failing samples.
- Use pRESTO Collapseseq with n=0 by default to improve runtime.

### CODE REVIEW

File: assets/multiqc_config.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 report_comment: >
+
   This report has been generated by the <a href="https://github.com/nf-core/airrflow/releases/tag/4.0" target="_blank">nf-core/airrflow</a>
```
Comment (ssnn-airr) [inline assets/multiqc_config.yml:3]: Should `tag/4.0` be `tag/4.2.0`?

File: README.md [hunk 2]
```diff
@@ -180,13 +181,14 @@ nf-core/airrflow was originally written by:
...
-- [Simon Heumos](@subwaystation)
+- [Alexander Peltzer](https://github.com/apeltzer)
 
 We thank the following people for their extensive assistance in the development of the pipeline:
 
 - [David Ladd](https://github.com/dladd)
-- [Friederike Hanssen](https://github.com/ggabernet/friederikehanssen)
+- [Friederike Hanssen](https://github.com/friederikehanssen)
+- [Simon Heumos](https://github.com/subwaystation)
+- [Mark Polster](https://github.com/mapo9)
```
Comment (mapo9) [inline README.md:191]: thanks for that :D

**Review by mapo9 — APPROVED** [review]
lgtm! 🚀

File: .nf-core.yml [hunk 3]
```diff
@@ -11,5 +12,17 @@ lint:
         - params.report_logo
         - params.report_logo_img
         - params.config_profile_url
+nf_core_version: 3.0.2
```
Comment (ggabernet) [inline .nf-core.yml:15]:
```suggestion
  included_configs: False
nf_core_version: 3.0.2
```

**Review by ssnn-airr — APPROVED** [review]
LGTM

**Comment by mapo9** [comment]
@ggabernet  I added the metromap directly to this pr. Please check it out

**Pull request merged**